addons/account_edi_ubl/models/account_edi_format.py

```diff
@@ -159,3 +157,85 @@ def _find_value(xpath, element=tree):
                             invoice_line_form.tax_ids.add(tax)
 
         return invoice_form.save()
+
+    ####################################################
+    # Export
+    ####################################################
+
+    def _get_ubl_values(self, invoice):
+        ''' Get the necessary values to generate the XML. These values will be used in the qweb template when
+        rendering. Needed values differ depending on the implementation of the UBL, as (sub)template can be overriden
+        or called dynamically.
+        :returns:   a dictionary with the value used in the template has key and the value as value.
+        '''
+        def format_monetary(amount):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
+            return float_repr(amount, invoice.currency_id.decimal_places)
```

Comment:
I think we need the currency.round as well in a lot of cases e.g. with the calculation of discounts.

Reply:
In my mind, every monetary contained in the values are already rounded

Reply:
Yes, but I started doubting with all the tax_detail_vals we send, but it seems to be rounded correctly there.  (at least what is in the xml)